python/ray/serve/controller.py

```diff
@@ -170,22 +168,61 @@ class Config:
 
 
 class BackendState:
-    def __init__(self, checkpoint: bytes = None):
+    def __init__(self,
+                 controller_name: str,
+                 detached: bool,
+                 checkpoint: bytes = None):
+        self.controller_name = controller_name
+        self.detached = detached
+
+        # Non-checkpointed state.
+        self.currently_starting_replicas: Dict[asyncio.Future, Tuple[
+            BackendTag, ReplicaTag, ActorHandle]] = dict()
+        self.currently_stopping_replicas: Dict[asyncio.Future, Tuple[
```

Comment:
Can you pull this into a new file?

Reply:
I will do this in a separate PR because it will make it very difficult to read the diff.

Reply:
We should also formalize the API for the `State` classes (not reach into internal fields in various places) which we can do at the same time.

Reply:
Sounds good, the main reason I wanted to do that in this PR is that git's diffing makes it look like EndpointState was changed, when I don't think that it was.